pkg/controllers/workspace_controller_test.go

```diff
@@ -786,14 +865,15 @@ func TestGetAllQualifiedNodes(t *testing.T) {
 
 				c.On("List", mock.IsType(context.Background()), mock.IsType(&corev1.NodeList{}), mock.Anything).Return(nil)
 			},
+			workspace:     test.MockWorkspaceWithPreferredNodes,
 			expectedError: nil,
+			expectedNodes: []string{"node1", "node-p1"},
```

Comment:
Unless overlooking something "node-p1" and "node-p3" look identical to me, why not node-p3 included here

Reply:
because "node-p3" is not in the preferredNodes field of testing case😊